lxd/storage/backend_lxd.go

```diff
@@ -3261,12 +3261,14 @@ func (b *lxdBackend) GetInstanceUsage(inst instance.Instance) (*VolumeUsage, err
 	volStorageName := project.Instance(inst.Project().Name, inst.Name())
 	vol := b.GetVolume(volType, contentType, volStorageName, dbVol.Config)
 
-	// Get the usage.
+	// Get the usage
+	// If storage driver does not support getting the volume usage, proceed getting the total.
 	size, err := b.driver.GetVolumeUsage(vol)
-	if err != nil {
+	if err != nil && !errors.Is(err, drivers.ErrNotSupported) {
 		return nil, err
 	}
 
+	// If driver does not support getting volume usage, this value would be -1.
```

Comment:
Sure!

Reply:
I was thinking a bit about this trying to identify what is LXD's tendency when it comes to displaying unbound entities. For example if you don't set `limits.cpu` for a container, you also don't see something like -1. 

Can't we just omit the `InstanceStateDisk.Usage/VolumeUsage.Used` fields using `omitempty` and only show the `.Total` in case usage=0/-1? This kind of follows the behavior of not displaying defaults.

Reply:
Maybe I wasn't precise enough. I didn't mean not showing `Total`. Always show `Total` but omit the `.Usage/.Used` fields for instance vols and custom storage vols if not supported. Then you don't have to worry about how to display the unbound case.

Reply:
That would be an API break right, because those fields arent pointers and so cant be nil/empty.

Reply:
Would it be an API break to only omit the output in the response JSON? I am talking about this pattern: (https://pkg.go.dev/encoding/json#Marshal)

```go
Usage string `json:"usage,omitempty"`
```

Then after performing unmarshal the client can check if `Usage == ""` it's not supported. 
If I understand the issue right currently the entire `InstanceStateDisk` struct isn't send if usage is not supported (-1).

Reply:
We are talking about `InstanceStateDisk` right, thats an int64 so cant be empty.

Reply:
> If I understand the issue right currently the entire InstanceStateDisk struct isn't send if usage is not supported (-1).

Thats another bug that needs fixing.

Reply:
Changing `Usage` from int64 to string is an API break.

Reply:
> We are talking about InstanceStateDisk right, thats Usage is an int64 so cant be empty.

Ah true, `omitempty` will also emit a `0` value but then it's not possible anymore to differentiate between "not supported" and actual "zero usage". Nvm then.